lib/streamlit/runtime/fragment.py

```diff
@@ -206,9 +206,16 @@ def wrapped_fragment():
                 )
                 with active_hash_context:
                     with st.container():
+                        # use dg_stack instead of active_dg to have correct copy during
+                        # execution (otherwise we can run into concurrency issues with
+                        # multiple fragments). Use dg_stack because we just entered a
+                        # container and [:-1] of the delta path because thats
+                        # the prefix of the fragment, e.g. [0, 3, 0] -> [0, 3]. All
+                        # fragment elements start with [0, 3].
+                        active_dg = dg_stack.get()[-1]
                         ctx.current_fragment_delta_path = (
                             active_dg._cursor.delta_path if active_dg._cursor else []
```

Comment:
nit: since the `active_dg` variable isn't really used much above, I think it would make sense to get rid of the name binding from above (and probably just access what we need directly from `dg_stack_snapshot` or name it something else so we don't have to shadow the variable here